core/network/impl/gossiper_broadcast.cpp

```diff
@@ -17,7 +17,75 @@ namespace kagome::network {
   void GossiperBroadcast::reserveStream(
       const libp2p::peer::PeerInfo &peer_info,
       std::shared_ptr<libp2p::connection::Stream> stream) {
-    streams_.emplace(peer_info, std::move(stream));
+    reserved_streams_.emplace(peer_info, std::move(stream));
+  }
+
+  void GossiperBroadcast::addStream(
+      std::shared_ptr<libp2p::connection::Stream> stream) {
+    auto peer_id_res = stream->remotePeerId();
+    if (not peer_id_res.has_value()) {
+      logger_->error("Can't get peer_id: {}", peer_id_res.error().message());
+      return stream->reset();
+    }
+    auto &peer_id = peer_id_res.value();
+
+    auto syncing_stream_it = syncing_streams_.find(peer_id);
+    if (syncing_stream_it != syncing_streams_.end()) {
+      if (syncing_stream_it->second != stream) {
```

Comment:
I'm not sure, that it is a good idea to compare streams by address of the object. More over the single peer can create multiple streams that will multiplex in a single data stream. Than we don't need to reset the existed.

Reply:
It will be refactored during other issue (#546)